src/app/shared/services/course.service.ts

```diff
@@ -3,28 +3,37 @@ import { HttpClient} from '@angular/common/http'
 import { BehaviorSubject, Observable, Subject } from 'rxjs';
 
 import { HttpCourse } from '../interfaces/http-course.interface';
+import { LoadingService } from './loading.service';
 
 @Injectable({
   providedIn: 'root'
 })
 export class CourseService {
 
   idCounter = 3;
-  private quanityOfVisibleCourses = 3;
+  inputSubject = new BehaviorSubject<string>('');
+  coursesSubject = new Subject<HttpCourse[]>();
   private apiUrl = 'http://localhost:3004';
-  inputSubject = new Subject<string>();
-  loadingSubject = new Subject<boolean>();
+  private quanityOfVisibleCourses = 3;
 
-  constructor(private http: HttpClient) {
-    this.inputSubject = new BehaviorSubject<string>('');
-   }
+  constructor(private http: HttpClient, private loadingService: LoadingService) {}
 
-  onFetchCourses(): Observable<HttpCourse[]> {
-    return this.http.get<HttpCourse[]>(`${this.apiUrl}/courses?start=0&count=${this.quanityOfVisibleCourses}`)
+  onFetchCourses() {
+    this.loadingService.loadingSubject.next(true);
+    this.http.get<HttpCourse[]>(`${this.apiUrl}/courses?start=0&count=${this.quanityOfVisibleCourses}`)
+      .subscribe(response => {
+        this.coursesSubject.next(response);
```

Comment:
Also remove the subscription for the service
You should handle the logic with the help of **RxJs** operators and then subscribe in the component

Reply:
done, shift the subscription to the component with the logic in the service